DB/Item.lua

```diff
@@ -150,4 +150,17 @@ function Item:IsMount(entry)
 	return entry.type == CONSTANTS.ITEM_TYPES.MOUNT
 end
 
+function Item:HasZone(entry)
+	if entry.coords then
+		for _, value in pairs(entry.coords) do
+			if type(value) == "table" and value.m ~= nil then
```

Comment:
Early exit?

``if type(waypointData) ~= "table" or not waypointData.m then return false end``

In fact, it may be more readable to split both conditions, or use some auxiliary name like ``local isValidEntry = <condition>`` and then ``if not isValidEntry then return end``? Not sure, just a few quick ideas.
The ``m`` is, of course, another awkward name, but that'll have to be changed individually (not part of this PR) :/

Reply:
The function had some unnecessary logic to it, so I rewrote most of it in 2b30bf11dc726546918879a62c0f240b8378f55f. It now has an early exit for any entries that are missing coords entirely, and it validates the item on the first occurrence of a mapID.